services/github/src/handlers.js

```diff
@@ -630,19 +630,42 @@ async function jobHandler(message) {
     if (!defaultBranchYml) { return; }
 
     if (this.getRepoPolicy(defaultBranchYml).startsWith('collaborators')) {
-      let login = message.payload.body.pull_request.head.user.login;
-
-      let isCollaborator = await instGithub.repos.checkCollaborator({
-        owner: organization,
-        repo: repository,
-        username: login,
-      }).catch(e => {
-        if (e.status !== 404) {
-          throw e;
-        }
-      });
-
-      if (!isCollaborator) {
+      // There are four usernames associated with a PR action:
+      //  - pull_request.user.login -- the user who opened the PR
+      //  - pull_request.head.user.login -- the username or org name for the repo from which changes are pulled
+      //  - pull_request.base.user.login -- the username or org name for the repo into which changes will merge
+      //  - sender.login -- the user who clicked the button to trigger this action
+      //
+      // The "collaborators" and "collaborators_quiet" policies require:
+      //  - pull_request.user.login is a collaborator; AND
+      //  - pull_request.head.user.login is
+      //    - a collaborator OR
+      //    - the same as pull_request.base.user.login
+      //
+      // Meaning that the PR must have been opened by a collaborator and be merging code from a collaborator
+      // or from the repo against which the PR is filed.
```

Comment:
so here we're relying on the repo's policies about who can push branches to it?

Reply:
Yes -- there was some discussion of that in the bug and it's not clear it's even possible to do better.